addons/controllers/clusterbootstrap_controller.go

```diff
@@ -1361,3 +1412,108 @@ func (r *ClusterBootstrapReconciler) reconcileClusterProxyAndNetworkSettings(clu
 
 	return nil
 }
+
+func (r *ClusterBootstrapReconciler) reconcileDelete(cluster *clusterapiv1beta1.Cluster, log logr.Logger) (ctrl.Result, error) {
+	remoteClient, err := util.GetClusterClient(r.context, r.Client, r.Scheme, clusterapiutil.ObjectKey(cluster))
```

Comment:
Suggestion: Can we move this statement up into `Reconcile` and pass remoteClient into `reconcileNormal` and `reconcileDelete` as parameter, so that we wouldn't need to call `util.GetClusterClient` twice in the code?